src/eynollah/eynollah.py

```diff
@@ -4522,467 +4519,471 @@ def run(self):
                 polygons_lines_xml = []
                 contours_tables = []
                 ocr_all_textlines = None
-
                 pcgts = self.writer.build_pagexml_no_full_layout(cont_page, page_coord, order_text_new, id_of_texts_tot,
                                                                  all_found_textline_polygons, page_coord, polygons_of_images, polygons_of_marginals,
                                                                  all_found_textline_polygons_marginals, all_box_coord_marginals, slopes, slopes_marginals,
                                                                  cont_page, polygons_lines_xml, contours_tables, ocr_all_textlines)
                 if self.dir_in:
+                    self.writer.write_pagexml(pcgts)
                     continue
                 else:
                     return pcgts
+            if not self.extract_only_images and not self.skip_layout_and_reading_order:
```

Comment:
This condition is completely redundant: both `extract_only_images` and `skip_layout_and_reading_order` had their condition checked and the function return by now, so you are essentially saying `if True` here.

Reply:
The huge diff in all remaining hunks (which could be much smaller if `diff --minimal` were used) is purely indentation AFAICS, correct?

(Okay, except for the two additional return tuple elements of `filter_contours_without_textline_inside`.)

Reply:
> The huge diff in all remaining hunks (which could be much smaller if `diff --minimal` were used) is purely indentation AFAICS, correct?
> 
> (Okay, except for the two additional return tuple elements of `filter_contours_without_textline_inside`.)

True. The only key things that were missed were extracting_only_images and skip_layout_ro.

Reply:
So I removed the redundant conditional and reduced the indentation of the remaining body of the function again.